sympy/simplify/tests/test_radsimp.py

```diff
@@ -342,6 +342,9 @@ def test_collect_const():
     assert collect_sqrt(eq + 2) == \
         2*sqrt(sqrt(2) + 3)*(sqrt(5)*x + y) + 2
 
+    # issue 16296
+    assert collect_const(a + b + x/2 + y/2) != a + b + (x/2 + y/2)
```

Comment:
Perhaps this could be `== a + b + Mul(S.Half, x + y, evaluate=False)` (though it does not look so nice).

Reply:
Yeah, I did not know how to stop evaluation, thanks.